src/include/storage/write_ahead_log/log_record.h

```diff
@@ -201,32 +215,40 @@ class DeleteRecord {
    *
    * @param head pointer location to initialize, this is also the returned address (reinterpreted)
    * @param txn_begin begin timestamp of the transaction that generated this log record
-   * @param table the data table this delete points to
+   * @param db_oid database oid of this delete record
+   * @param table_oid table oid of this delete record
    * @param slot the tuple slot this delete applies to
    * @return pointer to the initialized log record, always equal in value to the given head
    */
-  static LogRecord *Initialize(byte *const head, const transaction::timestamp_t txn_begin, DataTable *const table,
-                               TupleSlot slot) {
+  static LogRecord *Initialize(byte *const head, const transaction::timestamp_t txn_begin,
+                               const catalog::db_oid_t db_oid, const catalog::table_oid_t table_oid,
+                               const TupleSlot slot) {
     auto *result = LogRecord::InitializeHeader(head, LogRecordType::DELETE, Size(), txn_begin);
     auto *body = result->GetUnderlyingRecordBodyAs<DeleteRecord>();
-    body->table_ = table;
+    body->db_oid_ = db_oid;
+    body->table_oid_ = table_oid;
     body->tuple_slot_ = slot;
     return result;
   }
 
   /**
-   * @return pointer to the DataTable that this delete is concerned with
+   * @return the tuple slot changed by this delete record
+   */
+  TupleSlot GetTupleSlot() const { return tuple_slot_; }
+
+  /**
+   * @return database oid for this delete record
    */
-  DataTable *GetDataTable() const { return table_; }
+  catalog::db_oid_t GetDatabaseOid() const { return db_oid_; }
 
   /**
-   * @return the tuple slot changed by this delete record
+   * @return table oid for this delete record
    */
-  TupleSlot GetTupleSlot() const { return tuple_slot_; }
+  catalog::table_oid_t GetTableOid() const { return table_oid_; }
 
  private:
-  // TODO(Tianyu): Change to oid maybe?
-  DataTable *table_;
+  catalog::db_oid_t db_oid_;
+  catalog::table_oid_t table_oid_;
```

Comment:
I would be cautious about this change. The LogWriter still needs to know about the schema in order to write out varlens. Alternatively, you can change redo records to always inline varlens instead of using a pointer.

Reply:
The only thing the LogWriter needs is the `storage::BlockLayout` object from the `DataTable*`.

Reply:
RawBlocks store the DataTable* now, so in theory you can always get back to the DataTable from any TupleSlot. That should be enough to get BlockLayout, right?